### PR TITLE
Taymoor/feature/backend setup

### PR DESCRIPTION
**Pull request by taymoorKhalid**
Complete TodoList !

### CODE REVIEW

File: BackEnd/.gitignore
Comment (DanyalNazir10) [inline BackEnd/.gitignore:1]: Add EOF

File: BackEnd/controllers/task-controllers.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+};
+
+const deleteTask = async (req: Request, res: Response, next: NextFunction) => {
+  const { tid } = req.params;
+
+  try {
+    await Task.findByIdAndDelete(tid);
+
+    // Fetch remaining tasks for the user
```
Comment (DanyalNazir10) [inline BackEnd/controllers/task-controllers.ts]: Each api should serve a single purpose. Its a delete api so just use it to delete and if you want to get the updated record get the getTaskApi or something like that.

File: BackEnd/controllers/task-controllers.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    const updatedTask = await Task.findByIdAndUpdate(tid, updateData, {
+      new: true,
+    });
+
+    // If no task found, throw an error
+    if (!updatedTask) {
+      return next(new HttpError("Task not found.", 404));
+    }
+
+    // Fetch all tasks for the user (after the update)
```
Comment (DanyalNazir10) [inline BackEnd/controllers/task-controllers.ts]: Each api should serve a single purpose. Its a update api so just use it to update and if you want to get the updated record get the getTaskApi or something like that.

File: BackEnd/tsconfig.json
Comment (DanyalNazir10) [inline BackEnd/tsconfig.json:1]: If you don't need the commented rules, remove them.

File: FrontEnd/src/components/containers/SignupContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// containers/SignupContainer.tsx
+import { connect } from "react-redux";
+import Signup from "../../pages/Signup"; // Replace with the path to your Signup component
```
Comment (DanyalNazir10) [inline FrontEnd/src/components/containers/SignupContainer.tsx]: use proper comments if you have decided to use tem.

File: FrontEnd/src/components/containers/TodoItemContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { connect } from "react-redux";
+import {
+  // toggleTodoAction,
```
Comment (DanyalNazir10) [inline FrontEnd/src/components/containers/TodoItemContainer.tsx:3]: No commented code

File: FrontEnd/src/components/todoForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import Button from "../../shared/Button";
+import icons from "../../assets/svg/icons";
+
+import "./style.css";
+
+interface TodoFormProps {
+  addTodo: (todo: { text: string; isCompleted: boolean }) => void;
+}
+
+// Define Yup validation schema
```
Comment (DanyalNazir10) [inline FrontEnd/src/components/todoForm/index.tsx]: Create a separate folder as validations and define your forms validations there.

File: FrontEnd/src/components/todoItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  isCompleted: boolean;
+}
+
+interface TodoItemProps {
+  todo: editTodo;
+  onToggleTodo: (id: string) => void; // Pass ID to toggle
+  onDeleteTodo: (id: string) => void; // Pass ID to delete
+  onEditTodo: (id: string, newText: string) => void; // Pass ID to edit
+}
+
```
Comment (DanyalNazir10) [inline FrontEnd/src/components/todoItem/index.tsx:24]: Create a separate folder as validations and define your forms validations there.

File: FrontEnd/src/pages/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import icons from "../assets/svg/icons";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import Loader from "../shared/Loader";
+
+const schema = yup.object().shape({
```
Comment (DanyalNazir10) [inline FrontEnd/src/pages/Login.tsx]: Create a separate folder as validations and define your forms validations there.

File: FrontEnd/src/pages/Signup.tsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import { useNavigate } from "react-router-dom";
+import icons from "../assets/svg/icons";
+import InputField from "../shared/Input"; // Import the InputField component
+import "./signup.css";
+import Loader from "../shared/Loader";
+
+// Validation Schema
+const schema = yup.object().shape({
```
Comment (DanyalNazir10) [inline FrontEnd/src/pages/Signup.tsx]: Create a separate folder as validations and define your forms validations there.

File: FrontEnd/src/pages/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    mode: "onSubmit",
+  });
+  const navigate = useNavigate();
+
+  const onSubmit: SubmitHandler<LoginFormValues> = async (data) => {
+    onLogin(data.email, data.password);
+  };
+
+  return (
+    <div id="signup-wrapper">
```
Comment (DanyalNazir10) [inline FrontEnd/src/pages/Login.tsx:36]: You didn't used the common input field component here.

File: FrontEnd/src/store/reducers/authReducer.ts
Comment (DanyalNazir10) [inline FrontEnd/src/store/reducers/authReducer.ts:1]: You don't need to explicitly use the reducer again inn the file name. Its self explanatory. 
reducers/auth => it means its an auth reducer.
actions/auth => auth actions

don't repeat yourself by reducers/authReducer.
Do that for all the files where you are using this pattern.

File: FrontEnd/src/store/saga/sagas.ts
Comment (DanyalNazir10) [inline FrontEnd/src/store/saga/sagas.ts:1]: File name is not correct. It should be saga/todo

**Review by DanyalNazir10 — CHANGES_REQUESTED** [review]

File: BackEnd/controllers/task-controllers.ts (hunk 2 above)
Comment (taymoorKhalid) [reply BackEnd/controllers/task-controllers.ts]: So how should I do this ? from saga should I hit update or delete api first then do nothing with its response and then again hit the the getTaskAPI to get response and update state using reducers by passing its response to reducer?

**Pull request merged**